mount/mountinfo_linux_test.go

```diff
@@ -439,6 +439,9 @@ const (
 
 	mountInfoWithSpaces = `486 28 252:1 / /mnt/foo\040bar rw,relatime shared:243 - ext4 /dev/vda1 rw,data=ordered
 31 21 0:23 / /DATA/foo_bla_bla rw,relatime - cifs //foo/BLA\040BLA\040BLA/ rw,sec=ntlm,cache=loose,unc=\\foo\BLA BLA BLA,username=my_login,domain=mydomain.com,uid=12345678,forceuid,gid=12345678,forcegid,addr=10.1.30.10,file_mode=0755,dir_mode=0755,nounix,rsize=61440,wsize=65536,actimeo=1`
+
+	mountInfoWithDoubleQuotes = `1046 30 253:1 /tmp/bar /var/lib/kubelet/pods/98d150a4-d814-4d52-9068-b10f62d7a895/volumes/kubernetes.io~empty-dir/tmp-dir/"var rw,relatime shared:1 - ext4 /dev/mapper/ubuntu--vg-root rw,errors=remount-ro
```

Comment:
Please also test with a path that contains `\"` (backslash + quote)?

Reply:
Mounting a directory containing a backslash will display it accii escaped in `/proc/self/mountinfo`:

```
mkdir '/tmp/\"bar\"'
mkdir /tmp/bar
sudo mount --bind /tmp/bar '/tmp/\"bar\"'

cat /proc/self/mountinfo | grep bar
1100 30 253:1 /tmp/bar /tmp/\134"bar\134" rw,relatime shared:1 - ext4 /dev/mapper/ubuntu--vg-root rw,errors=remount-ro
```

So I added a testcase that receives the backslash ascii character. Did you mean it this way? It seems that a literal `\"` cannot show up in the output of `/proc/self/mountinfo`.

Reply:
Yes, thanks

Reply:
Just to double-check; the field (as a whole) is never quoted? (`"/var/lib/....."`), so no exception is needed for leading/trailing quote?

Reply:
oh (sorry for the noise); would single quotes have the same problem?

Reply:
No problem, valid questions :wink:. For the latter I added some more test cases.

Reply:
> It seems that a literal \" cannot show up in the output of /proc/self/mountinfo

It's not really clear to me then why it is using unquote at all

Reply:
> It's not really clear to me then why it is using unquote at all

So, mountinfo entries quote the following characters, using escape sequences with their octal ascii codes:
 * ` ` (i.e. space)  -- as `\040`
 * `\t` (i.e. tab) -- as `\011`
* `\n' (i.e. newline) -- as `\012`
* `\\` (i.e. backslash) -- as `\134`

The `unquote` was added (by @thaJeztah AFAIR) to convert these escape sequences back to spaces, tabs etc. It does that but messes up double quotes.

I think the solution is to manually interpret the above sequences.

Reply:
> I think the solution is to manually interpret the above sequences.

I mean, write our own function to de-escape `\040`, `\011`, `\012` and `\134` instead of using strconv.Unquite

Reply:
yeah, I agree with that. Looking at the implementation of `strconv.Unquote` makes it seem like it is unnecessary to quote then use that. If these are the case we need to test against, then we should add them to the tests.

If you look at the `strconv.Unquote` function you will see it first just does some checks whether it contains some of these special characters. This could probably do the same thing and if there is no backslash, just return.